src/rgw/rgw_common.h

```diff
@@ -2111,6 +2129,39 @@ struct rgw_cache_entry_info {
   rgw_cache_entry_info() : gen(0) {}
 };
 
+struct rgw_data_placement_volatile_config {
```

Comment:
This isn't a typical RGW name for a class/struct type.  Could we rename this?  More concretely, what makes it volatile, and how does that relate to similar info in RGW?

Reply:
To tell the truth, I'm also worry about the proper name for this struct. I just use it for now and prefer to change if a proper one is found.

“volatile” here means unlike index_pool, data_pool and data_extra_pool of a bucket, each object in a bucket may have different rgw_data_placement_volatile_config.
And if we change current_tail_pools and upload the same object again, rgw_data_placement_volatile_config of the same object will be changed.

Any recommendation?